cpp/daal/src/algorithms/objective_function/cross_entropy_loss/cross_entropy_loss_dense_default_batch_impl.i

```diff
@@ -573,19 +603,31 @@ services::Status CrossEntropyLossKernel<algorithmFPType, method, cpu>::compute(N
 
         DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION(size_t, n, sizeof(algorithmFPType));
 
-        TArrayScalable<algorithmFPType, cpu> aX(n * p);
-        TArrayScalable<algorithmFPType, cpu> aY(n);
+        if (_aX == nullptr || _aX->size() != n * p)
+        {
+            _aX = new TArrayScalable<algorithmFPType, cpu>(n * p);
```

Comment:
If `_aX != nullptr && _aX->size() != n * p` then memory leak happens. You need `_aX->resize()` or something like that. The same in Log Loss